src/lib/db/index.ts

```diff
@@ -152,12 +154,26 @@ export default class Database {
     return this.db().match2().where({ id: match2Id }).select(match2Columns)
   }
 
-  getLastProcessedBlock = async (): Promise<{ hash: string; parent: string; height: number } | null> => {
+  getLastProcessedBlock = async (): Promise<ProcessedBlock | null> => {
     const blockRecords = await this.db()
       .processed_blocks()
       .select(processBlocksColumns)
       .orderBy('height', 'desc')
       .limit(1)
     return blockRecords[0] || null
   }
+
+  insertProcessedBlock = async (block: ProcessedBlock): Promise<void> => {
+    await this.db().processed_blocks().insert(block)
+  }
+
+  withTransaction = (update: (db: Database) => Promise<void>) => {
+    return this.client.transaction(async (trx) => {
+      const decorated: Database = {
+        ...this,
+        client: trx,
+      }
+      await update(decorated)
+    })
```

Comment:
Note this is a little dicey. I would prefer client to be dependency injected so I could create a new `Database` here properly with the decorated client. That said it works quite gracefully becasue of how Typescript handles private class properties